python/desc/imsim/imSim.py

```diff
@@ -178,3 +202,32 @@ def extract_objects(df):
     phosim_galaxies = \
         phosim_galaxies.query("halfLightSemiMajor >= halfLightSemiMinor")
```

Comment:
I included it to be faithful to your code, but in fact, it isn't violated for the instance catalogs I checked, including the example from Scott.  I'm perfectly happy to remove it entirely.